tests/testthat/test-ML.NeuralNet.R

```diff
@@ -1 +1,115 @@
-context("ML.Neuralnet.R")
+context("ML.NeuralNet.R")
+described.class <- ML.NeuralNet
+
+context(" initialize")
+#=========================================================
+test_that("it should initialize", {
+  expect_error(described.class$new(), NA)
+})
+
+context(" do.fit")
+#==========================================================
+test_that("it should fit a basic model", {
+  data <- data.table(
+    x1=c(1,2,3,4,5,6,7,8,9,10),
+    x2=c(1,2,3,4,5,6,7,8,9,10)
+  )
+  y <- c(0,0,0,0,0,0,1,0,1,1)
+  subject <- described.class$new()
+  model <- subject$perform_fit(X_mat = data, Y_vals = y, save_model = FALSE)
+  expect_is(model, 'nn')
+})
+
+test_that("it should store a cache of the neural network", {
+  data <- data.table(
+    x1=c(1,2,3,4,5,6,7,8,9,10),
+    x2=c(1,2,3,4,5,6,7,8,9,10)
+  )
+  y <- c(0,0,0,0,0,0,1,0,1,1)
+  subject <- described.class$new()
+  file.remove(subject$get_file_name)
+  model <- subject$perform_fit(X_mat = data, Y_vals = y, save_model = TRUE)
+  expect_true(file.exists(subject$get_file_name))
+})
+
+context(" do.update")
+#==========================================================
+test_that("it should update the model and return the updated version",{
+  #make first a model and update this model
+  data <- data.table(
+    x1=c(1,2,3,4,5,6,7,8,9,10),
+    x2=c(1,2,3,4,5,6,7,8,9,10)
+  )
+  y <- c(0,0,0,0,0,0,1,0,1,1)
+  subject <- described.class$new()
+  model<-subject$perform_fit(X_mat = data, Y_vals = y)
+
+  data <- data.table(
+    x1=c(1,2,3,4,5,6,7,8,9,10),
+    x2=c(1,2,3,4,5,6,7,8,9,10)
+  )
+
+  y <- c(0,0,0,0,0,0,1,0,1,1)
+  subject <- described.class$new()
+  testfit<-subject$perform_update(X_mat = data, Y_vals = y, m.fit = model)
+  expect_is(testfit, 'nn')
+  expect_false(equals(testfit, model))
+  # TODO: Actually perform the test here
```

Comment:
It is important to actually perform the test at the end of the test file. What you would usually do is create a simple setup (like what you are doing now), and then at the end perform some kind of expectation. In this expectation you would then check whether the output of the functions you performed is what you'd actually expect. In the current form nothing is actually tested (apart from building it, and seeing if that would not generate any runtime errors). When you run the tests in the console, this should also generate some kind of warning (`Empty test`)

Reply:
See for example http://r-pkgs.had.co.nz/tests.html